tests/model_serving/model_runtime/vllm/toolcalling/test_granite_3_2_8b_instruct_preview.py

```diff
@@ -0,0 +1,201 @@
+            response_snapshot=response_snapshot,
+        )
+
+
+@pytest.mark.parametrize(
+    "model_namespace, s3_models_storage_uri, serving_runtime, vllm_inference_service",
+    [
+        pytest.param(
+            {"name": "granite-32-8b-multi"},
+            {"model-dir": MODEL_PATH},
+            {"deployment_type": KServeDeploymentType.RAW_DEPLOYMENT},
+            {
+                "deployment_mode": KServeDeploymentType.RAW_DEPLOYMENT,
+                "runtime_argument": SERVING_ARGUMENT,
+                "gpu_count": 2,
+                "name": "granite-32-8b-multi",
+                "min-replicas": 1,
+            },
+        ),
+    ],
+    indirect=True,
+)
+class TestGranite32ToolMultiModel:
+    def test_granite_multi_model_inference(
+        self,
+        vllm_inference_service: Generator[InferenceService, Any, Any],
+        get_pod_name_resource: Pod,
+        response_snapshot: Any,
+    ):
+        pod_name = get_pod_name_resource.name
+        model_info, chat_responses, completion_responses = run_raw_inference(
+            pod_name=pod_name,
+            isvc=vllm_inference_service,
+            port=8080,
+            endpoint=OPENAI_ENDPOINT_NAME,
+        )
+        model_info_tgis, completion_responses_tgis, completion_responses_tgis_stream = run_raw_inference(
+            pod_name=pod_name,
+            isvc=vllm_inference_service,
+            port=8033,
+            endpoint=TGIS_ENDPOINT_NAME,
+        )
+        validate_inference_output(
+            model_info,
+            chat_responses,
+            completion_responses,
+            model_info_tgis,
+            completion_responses_tgis,
+            completion_responses_tgis_stream,
+            response_snapshot=response_snapshot,
+        )
+
+    def test_granite_multi_model_tool_inference(
+        self,
+        vllm_inference_service: Generator[InferenceService, Any, Any],
+        get_pod_name_resource: Pod,
+        response_snapshot: Any,
+    ):
+        pod_name = get_pod_name_resource.name
+        model_info_w, chat_responses_w, completion_responses_w = run_raw_inference(
+            pod_name=pod_name,
+            isvc=vllm_inference_service,
+            port=8080,
+            endpoint=OPENAI_ENDPOINT_NAME,
+            chat_query=WEATHER_TOOL_QUERY,
+            tool_calling=WEATHER_TOOL[0],
+        )
+        model_info_l, chat_responses_l, completion_responses_l = run_raw_inference(
+            pod_name=pod_name,
+            isvc=vllm_inference_service,
+            port=8080,
+            endpoint=OPENAI_ENDPOINT_NAME,
+            chat_query=LIGHTSPEED_TOOL_QUERY,
+            tool_calling=LIGHTSPEED_TOOL[0],
+        )
+        validate_inference_output(
+            model_info_w,
+            chat_responses_w,
+            completion_responses_w,
+            model_info_l,
+            chat_responses_l,
+            completion_responses_l,
+            response_snapshot=response_snapshot,
+        )
```

Comment:
duplicate code, please extract and re-use

Reply:
Could you clarify what you mean by "duplicate code"? This function takes the output queries from the model's response and compares them with a variable for the model's response. While the response variable might be the same, the output differs. When you refer to "duplicate," what exactly do you mean? The purpose of this function is to handle multiple responses and match them accordingly. Also keeping here make it easy for the folks to understand better

Reply:
please see comment above